Controller Documentation

Holzhaus opened the pull request.

Holzhaus commented:
I something like this desired? I don't want to fix all this documentation if it's not desired anyway.

daschuer commented:
I think it is desired. But can we effort the maintenance burden? When I correctly remember it is already hard to keep the wiki updated.

Holzhaus commented:
> But can we effort the maintenance burden? When I correctly remember it is already hard to keep the wiki updated.

I only added controller that are included in Mixxx. 
I think it makes sense to have documentation for these mappings in the manual as the Wiki page quickly becomes tedious to read when the mapping changes between mixxx versions. We already have some mappings where the wiki page is just a changelog that is completely useless to the end user.

I think adding documentation to the manual opens the possibility to review the documentation properly, too. 

I don't think there's a big maintenance burden, we just need to make sure that anytime a controller mapping is changed the manual page is updated, too.

I suggest the following workflow:
1. Contributor opens PR for a controller mapping
2. We request to add a PR for documentation to the correct branch of this repo
3. We review the documentation and the mapping
4. We merge the mapping only if both the mapping and the documentation are considered ready to merge.

Holzhaus commented:
Ok, this is ready to review now. Please have a look.

All obvious formatting issues have been fixed. I removed most of the controller images because I doubt we have the copyright owner's permission and they also don't really add value. Schematic images and images with annotations have been kept.

I also removed all programming related info (MIDI value documentation, `lsusb` output, etc.), because that belongs into the wiki. Changelogs, review links, etc. have been removed, too.

Be-ing commented:
How does https://github.com/mixxxdj/mixxx/wiki/Hardware%20compatibility fit into this? Should this be moved into the manual?

Holzhaus commented:
> How does https://github.com/mixxxdj/mixxx/wiki/Hardware%20compatibility fit into this? Should this be moved into the manual? I am unsure about that because it lists prices which can change more frequently than we make releases.

I don't think we should put current street prices into the manual. We could use the MSRP or just keep it as simple list (i.e. keep the [current "Controllers" page as-is](https://deploy-preview-122--mixxx-manual.netlify.app/chapters/controllers.html)). The manual is for looking up how your controller works with Mixxx, not for using it as a purchase advice guide.

Be-ing commented:
We could make https://github.com/mixxxdj/mixxx/wiki/Hardware%20compatibility link to the manual pages for controllers instead of wiki pages.

Holzhaus commented:
Either that or we keep the wiki pages and replace the mapping description with a link to the manual page. Maybe the latter is preferable because some wiki pages contain useful information for mapping developer that I didn't include here (e.g. MIDI message documentation). Also, old Mixxx version include a link to the wiki page, so maybe we shouldn't remove it.

Be-ing commented:
Sounds good to me

Holzhaus commented:
Ok. Shall we merge?

Holzhaus commented:
Merge?

Holzhaus commented:
@daschuer Merge?

daschuer commented:
The wiki link links to the wiki main page. I guess an average user is pretty lost looking for community mappings. 
I think we need to add some words how to search for them and where. 
The standard GitHub Search feature is not very helpful IMHO.

daschuer commented on source/chapters/controllers/behringer_cmd_mm_1.rst:
```diff
@@ -0,0 +1,229 @@
+Behringer CMD MM-1
+==================
+
+.. figure:: ../../_static/controllers/behringer_cmd_mm_1_overview.png
```
This is a 7,2 MB picture loading slooooow

daschuer commented on source/chapters/controllers/behringer_cmd_studio_4a.rst:
```diff
@@ -0,0 +1,192 @@
+stereo headphones) USB sound card built in.
+
+-  `Mixxx Forum Thread <http://www.mixxx.org/forums/viewtopic.php?f=7&t=7868>`__
+-  `Manufacturer’s product page <http://www.music-group.com/Categories/Behringer/Computer-Audio/DJ-Controllers/CMD-STUDIO-4a/p/P0809/Features>`__
```
dead

daschuer commented on source/chapters/controllers/denon_mc6000mk2.rst:
```diff
@@ -0,0 +1,120 @@
+Denon MC6000MK2
+===============
+
+-  `Manufacturer’s product page <http://denondj.com/products/view/mc6000mk2>`__
```
dead

daschuer commented on source/chapters/controllers/denon_sc2000.rst:
```diff
@@ -0,0 +1,9 @@
+Denon SC2000
+============
+
+-  `Manufacturer’s product page <http://denondj.com/products/view/dn-sc2000>`__
```
dead

daschuer commented on source/chapters/controllers/gemini_firstmix.rst:
```diff
@@ -0,0 +1,9 @@
+Gemini FirstMix
+===============
+
+-  `Manufacturer’s product page <http://geminisound.com/product/firstmix>`__
```
dead

daschuer commented on source/chapters/controllers/hercules_dj_console_4_mx.rst:
```diff
@@ -0,0 +1,319 @@
+
+The Hercules DJ Console 4-Mx is a USB controller with a built in sound card. It is very similar to the [[Hercules DJ Console RMX]], but with switches for 4 decks.
+
+-  `Manufacturer’s product page <http://www.hercules.com/us/DJ-Music/bdd/p/141/dj-console-4-mx/>`__
```
dead

daschuer commented on source/chapters/controllers/hercules_dj_console_rmx.rst:
```diff
@@ -0,0 +1,258 @@
+Hercules DJ Console RMX
+=======================
+
+-  `Manufacturer’s product page <http://www.hercules.com/us/legacy/bdd/p/62/dj-console-rmx/>`__
```
dead

daschuer commented on source/chapters/controllers/hercules_dj_console_rmx_2.rst:
```diff
@@ -0,0 +1,212 @@
+Hercules DJ Console RMX 2
+=========================
+
+-  `Manufacturer’s product page <http://www.hercules.com/us/DJ-Music/bdd/p/193/djconsole-rmx-2/>`__
```
dead

daschuer commented on source/chapters/controllers/hercules_djcontrol_air.rst:
```diff
@@ -0,0 +1,107 @@
+vu-meter.
+
+-  `Forum Thread <https://www.mixxx.org/forums/viewtopic.php?f=7&t=3263&p=25704&hilit=air#p25704>`__
+-  `Manufacturer’s product page <http://www.hercules.com/fr/legacy/bdd/p/174/djcontrol-air/>`__
```
dead

daschuer commented on source/chapters/controllers/hercules_djcontrol_compact.rst:
```diff
@@ -0,0 +1,97 @@
+Hercules DJ Control Compact
+===========================
+
+-  `Manufacturer’s product page <https://www.hercules.com/us/DJ-Music/bdd/p/253/djcontrol-compact/>`__
```
dead

daschuer commented on source/chapters/controllers/hercules_djcontrol_instinct.rst:
```diff
@@ -0,0 +1,95 @@
+
+-  Manufacturer’s product pages:
+
+   -  `DJ Control Instinct S <http://www.hercules.com/us/leisure-controllers/bdd/p/248/djcontrol-instinct-s-series/>`__
```
dead

daschuer commented on source/chapters/controllers/hercules_djcontrol_instinct.rst:
```diff
@@ -0,0 +1,95 @@
+-  Manufacturer’s product pages:
+
+   -  `DJ Control Instinct S <http://www.hercules.com/us/leisure-controllers/bdd/p/248/djcontrol-instinct-s-series/>`__
+   -  `DJ Control Instinct <http://www.hercules.com/us/DJ-Music/bdd/p/187/djcontrol-instinct/>`__
```
dead

daschuer commented on source/chapters/controllers/hercules_p32_dj.rst:
```diff
@@ -0,0 +1,101 @@
+
+   https://www.hercules.com/wp-content/uploads/2018/05/HerculesDJ_HerculesP32DJ_top-1400x747.jpg
+
+-  `Manufacturer’s product page <http://www.hercules.com/uk/advanced-controllers/bdd/p/258/hercules-p32-dj/>`__
```
dead

daschuer commented on source/chapters/controllers/ion_discover_dj.rst:
```diff
@@ -0,0 +1,57 @@
+The ION Discover DJ is a cheap budget MIDI controller meant to “discover DJ-ing” for people with a potential interest. It does not have a built in sound card, so a `splitter
+cable <hardware%20compatibility#splitter%20cables>`__ or `separate sound card <hardware%20compatibility#USB%20sound%20cards>`__ is required to use headphones with it.
+
+-  `Manufacturer’s product page <http://www.ionaudio.com/products/details/discover-dj>`__
```
dead

daschuer reviewed (changes requested):
Many external links are dead. I have checked only the first.

Holzhaus commented on source/chapters/controllers/akai_lpd8.rst:
```diff
@@ -0,0 +1,9 @@
+
+.. note::
+   Unfortunately a detailed description of this controller mapping is still missing.
+   If you own this controller, please consider contributing one.
```
What contributing guide? Do we have a guide for writing controller mapping documentation?

daschuer replied:
I think not explicit. Maybe we place two sentences somewhere and link to them here.

Holzhaus commented on source/chapters/controllers.rst:
```diff
@@ -0,0 +1,16 @@
+`Controller Mapping Forum <https://mixxx.discourse.group/c/controller-mappings/10>`__ and the
+`Wiki <https://github.com/mixxxdj/mixxx/wiki>`__.
+
+.. toctree::
```
AFAIK that's not possible. Sub-toctrees inherit the numbering from their parent, and I see no way to disable this.

daschuer commented on source/chapters/controllers/reloop_jockey_3_master_edition.rst:
```diff
@@ -0,0 +1,156 @@
+special drivers on Windows and macOS. *There is no driver available for Linux.*
+
+-  `Mixxx Forum Thread <https://mixxx.discourse.group/t/create-a-mapping-for-reloop-jockey-3-me/13703>`__
+-  `Manufacturer’s product page <http://www.reloop.com/reloop-jockey-3-me>`__
```
Dead

daschuer commented on source/chapters/controllers/tascam_us_428.rst:
```diff
@@ -0,0 +1,10 @@
+Tascam US-428
+=============
+
+-  `Manufacturer’s product page <http://www.tascam.com/products/us-428.html>`__
```
Dead

daschuer reviewed (changes requested):
Some more broken links

Holzhaus commented on source/chapters/controllers/native_instruments_traktor_kontrol_s4_mk2.rst:
```diff
@@ -0,0 +1,113 @@
+The Kontrol S4 Mk2 can run from USB bus power. Using the separate power supply increases the brightness of the LEDs, which is helpful for using it in daylight, and increases the volume of the
+headphone output.
+
+-  `Manufacturer’s product page <https://www.native-instruments.com/en/products/traktor/dj-controllers/traktor-kontrol-s4/>`__
```
No, it doesn't. Maybe there was a bug on the NI webserver?

Holzhaus commented:
Should we rename the "controllers" section to "hardware"? That way we leave the door open for non-controller hardware, e.g. Soundcards, DVS hardware, etc.

Holzhaus commented:
I vastly improved the Roland DJ-505 mapping description (didn't bother to do it in the wiki).

daschuer reviewed (approved):
LGTM now. Thank you.

How do we want to proceed?
Delete the wiki pages and redirect to the manual? 

Before, we need to make sure contributors are aware how to improve the controller manual and which part is mandatory when contributing a new mapping.

Holzhaus commented:
> LGTM now. Thank you.
> 
> How do we want to proceed?
> Delete the wiki pages and redirect to the manual?

Yes.

> Before, we need to make sure contributors are aware how to improve the controller manual and which part is mandatory when contributing a new mapping.

https://github.com/mixxxdj/mixxx/wiki/Contributing-Mappings#documenting-the-mapping

daschuer commented:
Ok, so we can merge now?

Holzhaus commented:
Yes.

daschuer commented:
Ok, thank you.
